server/src/score/workshop-score.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Application } from '../application/application.entity';
+import { GenericException } from '../common/generic-exception';
+import { User } from '../user/user.entity';
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { ScoreDto } from './dto/score.dto';
+import { ScoreError } from './score.errors';
+import { WorkshopScore } from './workshop-score.entity';
+
+@Injectable()
+export class WorkshopScoreService {
+  constructor(
+    @InjectRepository(WorkshopScore)
+    private workshopScoreRepository: Repository<WorkshopScore>
+  ) {}
+
+  async getWorkshopScores(applicationId: string) {
+    return await this.workshopScoreRepository.find({
+      where: { application: applicationId },
+      relations: ['user'],
+      loadRelationIds: true,
+    });
+  }
+
+  async getWorkshopScore(id: string) {
+    const score = await this.workshopScoreRepository.findOne(id, {
+      relations: ['user', 'application'],
+    });
+    if (!score) {
+      throw new GenericException(ScoreError.SCORE_NOT_FOUND);
+    }
+    return score;
+  }
+
+  async createWorkshopScore(
+    user: User,
+    application: Application,
+    scoreDto: ScoreDto
+  ): Promise<WorkshopScore> {
+    const score = await this.workshopScoreRepository.create(scoreDto);
+    score.user = user;
+    score.application = application;
+
+    return this.workshopScoreRepository.save(score);
```

Comment:
This would throw an error if there's an Workshop score entry in the table! So the code should check if a workshop score exists for the application and throw a ScoreError!